specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-03-01-preview/diagnostic.json

```diff
@@ -0,0 +1,694 @@
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the metadata of Spot Placement Recommender diagnostic.": {
+            "$ref": "./examples/GetSpotPlacementRecommender.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/spotPlacementRecommender/generate": {
+      "post": {
+        "tags": [
+          "spotPlacementRecommender"
+        ],
+        "operationId": "SpotPlacementRecommender_Post",
+        "description": "Generates placement scores for Spot VM skus.",
+        "parameters": [
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "spotPlacementRecommenderInput",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SpotPlacementRecommenderInput"
+            },
+            "description": "SpotPlacementRecommender object supplied in the body of the Post spot placement recommender operation."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/SpotPlacementRecommenderResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Returns spot VM placement scores for given configurations.": {
+            "$ref": "./examples/SpotPlacementRecommender.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ComputeDiagnosticsList": {
+      "description": "Lists all available Compute diagnostics for a subscription in a location.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ComputeDiagnosticBase"
+          },
+          "description": "The collection of available Compute diagnostics returned by the listing operation."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The continuation token."
+        }
+      }
+    },
+    "ComputeDiagnosticBase": {
```

Comment:
>ComputeDiagnosticBase [](http://example.com/codeflow?start=5&length=21)

can you please define this with allof Proxy resource from common types ?

Reply:
Hi Suhas, could you help point me to any examples/documentation on how to do this? Thanks

Reply:
In this file you have already done this for things like default errorresponse correctly on line 394 : 


Similar to that the COmputeDiagnostics is a Proxy resource as defined here : [https://github.com/Azure/azure-rest-api-specs/blob/9e7fb3b27c6b744746a5efd8d04f42efab17bc7a/specification/common-types/resource-management/v5/types.json#L92](https://github.com/Azure/azure-rest-api-specs/blob/9e7fb3b27c6b744746a5efd8d04f42efab17bc7a/specification/common-types/resource-management/v5/types.json#L92)

This is the base class for all Proxy resources in azure.

Reply:
you should be able to search in this repo for references to the ProxyREsource from commontypes for examples

Reply:
Hi Suhas, thanks for the reply. I tried updating this commit: https://github.com/Azure/azure-rest-api-specs/pull/27899/commits/14d18a17eabf9a35d67501a5836efe34d747ec21

But the model validation is failing because the ID in ComputeDiagnosticBase is not an ARM ID but our own service's ID for the diagnostic name:

![image](https://github.com/Azure/azure-rest-api-specs/assets/18197375/f361233e-90de-4bcc-8a15-388d91572d79)

Reply:
name is a top level ARM property it must be fully qualified 
    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection": {

You are not free to change the format for arm level properties, only within the property bag.

Reply:
Could you also respond to my other comment about the resources : https://github.com/Azure/azure-rest-api-specs/pull/27899#issuecomment-1989308792

Reply:
Updated the examples for ComputeDiagnosticBase

![image](https://github.com/Azure/azure-rest-api-specs/assets/18197375/0eba0556-adec-4c30-b472-3b3fee298c29)

Reply:
so can you now change the type also for "ComputeDiagnosticBase" to be a ProxyResource ? Also please respond to my other comment. You can setup an API modelling office hours if you need to discuss this in person  : https://eng.ms/docs/products/arm/introduction/office_hours

Reply:
Hi Suhas, sorry I forgot to include that change, I have update the PR now, and my team member @ansahdev has replied to the other comment further down in the thread, please let us know your feedback and we can discuss in office hours if necessary. However, the earliest available office hours is next week whereas we would prefer to have feedback on this PR sooner if possible. Thanks for your understanding